pkg/metrics/observers/osm.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2021 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package observers
+
+import (
+	"fmt"
+	"time"
+
+	flaggerv1 "github.com/fluxcd/flagger/pkg/apis/flagger/v1beta1"
+	"github.com/fluxcd/flagger/pkg/metrics/providers"
+)
+
+// TODO ask Stefan whether response_code=~"20." is a more suitable measurement for request success
+var osmQueries = map[string]string{
+	"request-success-rate": `
+    sum(
+        rate(
+            osm_request_total{
+				destination_namespace="test",
+				destination_kind="Deployment",
+				destination_name="podinfo",
```

Comment:
We need to use vars here, please see any other implementation.

Reply:
Resolved in https://github.com/fluxcd/flagger/pull/955/commits/5282707bb02ea46fa0e938586cf96c4d83ba3352

Reply:
Done.

https://github.com/fluxcd/flagger/pull/955/files#diff-2cf8c3821fd3970d6d29c1e9d0364bf70b88553caa2530889da1448517a6baf2R31-R59